storm-client/src/jvm/org/apache/storm/scheduler/resource/strategies/scheduling/DefaultResourceAwareStrategy.java

```diff
@@ -567,9 +567,9 @@ public int compare(Component o1, Component o2) {
             public int compare(Component o1, Component o2) {
                 int connections1 = o1.execs.size() * thisComp.execs.size();
                 int connections2 = o2.execs.size() * thisComp.execs.size();
-                if (connections1 > connections2) {
```

Comment:
hey guys this change is not right.  This is suppose to sort by the number of connections in descending order so that tasks with most connections to each other can be attempted to be colocated first.  What is the reason for this change?